.travis.yml

```diff
@@ -34,27 +34,51 @@ dist: trusty
 language: python
 git:
   depth: 1000
-before_install: 
+before_install:
 addons:
+  hostname: mtt_host
   apt:
     packages:
       - doxygen
       - doxygen-doc
       - doxygen-latex
       - doxygen-gui
       - graphviz
+
+      # install MPI libs
+      - openmpi-bin
+      - mpich
+      - libmpich-dev
+      - libopenmpi-dev
+
+      # install SLURM
+      - slurm-llnl
 install:
   - pip install -r pyenv.txt
 script:
+  - export TRAVIS_BUILD_DIR=$(pwd)
+  - export KERNEL_VERSION=4.4.0-101-generic
+
+  # symlink so BAT w/ hardcoded paths still run
+  - mkdir -p /opt/mtt/tests/
+  - ln -s $TRAVIS_BUILD_DIR/tests/bat /opt/mtt/tests/bat
+
+  # single-node BAT [order matters for last two]
+  - $TRAVIS_BUILD_DIR/pyclient/pymtt.py --verbose --base-dir=$TRAVIS_BUILD_DIR/pylib $TRAVIS_BUILD_DIR/tests/bat/negative.ini
+  - $TRAVIS_BUILD_DIR/pyclient/pymtt.py --verbose --base-dir=$TRAVIS_BUILD_DIR/pylib $TRAVIS_BUILD_DIR/tests/bat/default_collect_profile.ini
+  - $TRAVIS_BUILD_DIR/pyclient/pymtt.py --verbose --base-dir=$TRAVIS_BUILD_DIR/pylib $TRAVIS_BUILD_DIR/tests/bat/default_check_profile.ini
+
+before_deploy:
   - cd docs && doxygen Doxyfile
 deploy:
   provider: pages
   skip_cleanup: true
   github_token: $GITHUB_TOKEN
-  local_dir: $TRAVIS_BUILD_DIR/docs/html
+  local_dir: docs/
```

Comment:
I don't know Travis at all - is there some place I should see where it creates pages in the gh-pages branch per the title on this PR?

Reply:
I'm also curious why this change at line 54.

Reply:
The local-dir flag uploads the contents of that directory to the gh-pages branch. But it assumes a path relative to the current directory, so giving the absolute path made Travis complain because it kept wanting to change dir to an invalid path (i.e. /home/travis/build/Jagaskak/mtt/home/travis/build/Jagaskak/mtt/docs/html)

Travis also happens to overwrite the contents of the gh-pages branch when you upload a file, which is why I added the files to build the website to the docs folder and uploaded the entire file. 

On line 55, which you can't see here, I set the keep-history flag to true, so that Travis doesn't overwrite the history of the branch as well.